core/jvm/src/test/scala/scalaz/zio/RTSSpec.scala

```diff
@@ -327,6 +330,16 @@ class RTSSpec(implicit ee: ExecutionEnv) extends Specification with AroundTimeou
   def testRaceOfValueNever =
     unsafePerformIO(IO.point(42).race(IO.never[Throwable, Int])) == 42
 
+  def testRaceAllOfValues =
+    unsafePerformIO(IO.raceAll[Int, Int](List(IO.fail(42), IO.now(24))).attempt) == Right(24)
+
+  def testRaceAllOfFailures =
+    unsafePerformIO(IO.raceAll[Int, Void](List(IO.fail(42), IO.fail(24))).attempt) == Left(24)
```

Comment:
When you race two IOs they are forked, so here you don't have any guaranties about the finishing order, hence you don't know which one will be the last. I'd delay the one you want to be the last.

Reply:
Gotcha 👍 I had thought that was baked into the semantics of `race`. And a note to myself to read the RTS implementation, otherwise I'll keep on making wrong assumptions.